src/sc2laddercore/LadderConfig.cpp

```diff
@@ -1,76 +1,67 @@
+#include "LadderConfig.h"
+
+#include "Types.h"
+
+#include "ostreamwrapper.h"
+#include "writer.h"
+#include "prettywriter.h"
+
 #include <string>
 #include <map>
 #include <iostream>
 #include <sstream>
-#include <fstream>      
-#include "LadderConfig.h"
+#include <fstream>    
 
-LadderConfig::LadderConfig(const std::string &InConfigFile)
-	:ConfigFileLocation(InConfigFile)
+
+LadderConfig::LadderConfig(const std::string &ConfigFile)
+	:ConfigFileLocation(ConfigFile)
 {
+	// Doing this allows us to write values to the doc
+	// straight off instead of loading from file
+	this->doc.SetObject();
 }
 
 bool LadderConfig::ParseConfig()
 {
-	std::ifstream ifs(ConfigFileLocation, std::ifstream::binary);
-	if (!ifs)
-	{
-		return false;
-	}
-	
-
-	std::string line;
-	while (std::getline(ifs, line))
+	std::ifstream t(this->ConfigFileLocation);
+	if (t.good())
 	{
-		std::istringstream is_line(line);
-		std::string key;
-		if (std::getline(is_line, key, '='))
-		{
-			std::string value;
-			if (std::getline(is_line, value))
-			{
-				TrimString(key);
-				TrimString(value);
-				options.insert(std::make_pair(key, value));
-			}
-		}
+		std::stringstream buffer;
+		buffer << t.rdbuf();
+		std::string ConfigString = buffer.str();
+		return !doc.Parse(ConfigString.c_str()).HasParseError();
 	}
-	return true;
+	else
+		return false;
 }
 
+
 bool LadderConfig::WriteConfig()
 {
-	std::ofstream outfile(ConfigFileLocation, std::ofstream::binary);
-	for (const auto& setting : options)
-	{
-		outfile << setting.first << "=" << setting.second << std::endl;
-	}
-	outfile.close();
-	return true;
+	std::ofstream ofs(this->ConfigFileLocation.c_str());
+	rapidjson::OStreamWrapper osw(ofs);
+	rapidjson::PrettyWriter<rapidjson::OStreamWrapper> writer(osw);
+	return this->doc.Accept(writer);
 }
 
 std::string LadderConfig::GetValue(std::string RequestedValue)
 {
-	auto search = options.find(RequestedValue);
-	if (search != options.end())
-	{
-		return search->second;
-	}
-	return std::string();
-}
-
-void LadderConfig::TrimString(std::string &Str)
-{
-	size_t p = Str.find_first_not_of(" \t");
-	Str.erase(0, p);
-	p = Str.find_last_not_of(" \t\r\n");
-	if (std::string::npos != p)
+	if (doc.HasMember(RequestedValue))
 	{
-		Str.erase(p + 1);
+		if (doc[RequestedValue].IsString())
+			return doc[RequestedValue].GetString();
```

Comment:
Should always have braces on an if statement, just makes things a lot cleaner and easier to read

Reply:
This one is fixed.